lib/erb.rb

```diff
@@ -548,7 +549,7 @@ class Buffer # :nodoc:
       def initialize(compiler, enc=nil)
         @compiler = compiler
         @line = []
-        @script = enc ? "#coding:#{enc}\n" : ""
+        @script = enc ? "#coding:#{enc}\n".dup : String.new
```

Comment:
You remember correctly. But perhaps it would be a bit clearer to use just one syntax for creating mutable strings – I personally prefer `String.new` over calling `dup` on a newly created (immutable) string.

I’d suggest this line changing to either of these:
```rb
@script = enc ? String.new("#coding:#{enc}\n") : String.new

# or

@script = String.new
@script << "#coding:#{enc}\n" if enc
```